opentelemetry-sdk/src/opentelemetry/sdk/util/instrumentation.py

```diff
@@ -82,42 +82,56 @@ class InstrumentationScope:
     properties.
     """
 
-    __slots__ = ("_name", "_version", "_schema_url")
+    __slots__ = ("_name", "_version", "_schema_url", "_attributes")
 
     def __init__(
         self,
         name: str,
         version: Optional[str] = None,
         schema_url: Optional[str] = None,
+        attributes: Optional[dict] = None,
     ) -> None:
         self._name = name
         self._version = version
         if schema_url is None:
             schema_url = ""
         self._schema_url = schema_url
+        self._attributes = attributes
```

Comment:
I'm wondering if it makes sense for this to be an immutable structure. According to the spec, `Since the scope is a build-time concept, the attributes of the scope cannot change at runtime.`. Maybe use BoundedAttributes here?

Reply:
Like it has been done in this PR: https://github.com/open-telemetry/opentelemetry-python/pull/3310/files?

Reply:
I have resolved this, please review again. thanks!